lib/api/client-commands/registerBasicAuth.js

```diff
@@ -0,0 +1,50 @@
+const ClientCommand = require('./_base-command.js');
+const {Logger} = require('../../utils');
+
+/**
+ * Automate the input of basic auth credentials whenever they arise.
+ *
+ * @example
+ *  this.demoTest = function (browser) {
+ *    browser.registerBasicAuth(1000, 800);
+ *    browser.url('https://page-needs-auth-credentials.com');
+ *  };
+ *
+ *
+ * @method registerBasicAuth
+ * @param {string} username
+ * @param {string} password
+ * @param {function} [callback] Optional callback function to be called when the command finishes.
+ */
+class RegisterBasicAuth extends ClientCommand {
+
+  performAction(callback) {
+
+    const {browserName}  = this.api.capabilities;
+    if (browserName!=='chrome' && browserName!=='edge') {
```

Comment:
I think firefox also supports CDP, as I went through the code of selenium and found we can enable firefox CDP.
Ref: https://github.com/SeleniumHQ/selenium/blob/941da9bc1a2c18737e86ea58afcb89050239804d/javascript/node/selenium-webdriver/firefox.js#L419-L425

Reply:
Also, I went through the code base of selenium, and we can use `getCapabilities` method to check if we have CDP in the driver.
Ref: 
1. https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_Builder.html#getCapabilities 
2. https://github.com/SeleniumHQ/selenium/blob/944aad294d14a13345ebaf0e64df3aa68218521f/javascript/node/selenium-webdriver/lib/webdriver.js#L1200-L1207

Reply:
`firefox` supports CDP but does not suppport register basic authentication

Reply:
But maybe we can extra check for CDP from capabilities.